src/main/java/org/sagebionetworks/web/client/widget/entity/DoiWidget.java

```diff
@@ -35,75 +32,56 @@
 	
 	@Inject
 	public DoiWidget(DoiWidgetView view,
-			SynapseClientAsync synapseClient,
 			GlobalApplicationState globalApplicationState, 
 			StackConfigServiceAsync stackConfigService,
 			AuthenticationController authenticationController) {
 		this.view = view;
 		this.view.setPresenter(this);
-		this.synapseClient = synapseClient;
 		this.stackConfigService = stackConfigService;
 		this.globalApplicationState = globalApplicationState;
 		this.authenticationController = authenticationController;
 	}
 	
-	public void configure(String entityId, boolean canEdit, Long versionNumber) {
-		this.entityId = entityId;
-		this.versionNumber = versionNumber;
-		this.canEdit = canEdit;
-		configureDoi();
-	}
-	
 	public Widget asWidget() {
 		return view.asWidget();
 	}
-
-	public void configureDoi() {
+	
+	public void configure(Doi newDoi) {
 		clear();
-		//get this entity's Doi (if it has one)
-		doi = null;
-		timer = null;
-		synapseClient.getEntityDoi(entityId, versionNumber, new AsyncCallback<Doi>() {
-			@Override
-			public void onSuccess(Doi result) {
-				doi = result;
-				final DoiStatus doiStatus = doi.getDoiStatus();
-				if (doiStatus == DoiStatus.ERROR) {
-					view.showDoiError();
-				} else if (doiStatus == DoiStatus.IN_PROCESS) {
-					view.showDoiInProgress();
-				} else if (doiStatus == DoiStatus.CREATED || doiStatus == DoiStatus.READY) {
-					getDoiPrefix(new AsyncCallback<String>() {
-						@Override
-						public void onSuccess(String prefix) {
-							view.showDoiCreated(getDoi(prefix, doiStatus == DoiStatus.READY));
-						}
-						
-						@Override
-						public void onFailure(Throwable caught) {
-							if(!DisplayUtils.handleServiceException(caught, globalApplicationState, authenticationController.isLoggedIn(), view))
-								view.showErrorMessage(caught.getMessage());
-						}
-					});
-				}				
-				if ((doiStatus == DoiStatus.IN_PROCESS) && timer == null) {
-					timer = new Timer() {
-						public void run() {
-							configureDoi();
-						};
+		if (newDoi == null) {
+			view.showErrorMessage("No DOI found!");
```

Comment:
This widget used to show nothing if no DOI was found (NotFoundException).  With these changes, it looks like an error message will always pop up if an entity does not have a DOI.  If this is true, then it should be changed to the previous behavior (because many entities do not have DOIs, and that's ok).